cpu/cc26x0/cpu.c

```diff
@@ -17,13 +17,47 @@
  * @}
  */
 
-#include "cpu.h"
+ #include "cpu.h"
+ #include "periph_conf.h"
 
-/**
- * @brief Initialize the CPU, set IRQ priorities
- */
-void cpu_init(void)
-{
-    /* initialize the Cortex-M core */
-    cortexm_init();
-}
+ static void cpu_clock_init(void);
+
+ /**
+  * @brief Initialize the CPU, set IRQ priorities
+  */
+ void cpu_init(void)
+ {
+     /* initialize the Cortex-M core */
+     cortexm_init();
+
+     /*initialize the system clock*/
+     cpu_clock_init();
+ }
+
+ /**
+   *@brief Configure the MCU system clock
+   */
+ static void cpu_clock_init(void)
+ {
+   AON_WUC->AUXCTL |= AUXCTL_AUX_FORCE_ON; /* power on AUX_PD */
+   while(!(AON_WUC->PWRSTAT & PWRSTAT_AUX_PD_ON)); /* wait for AUX_PD to be powered on */
+   AUX_WUC->MODCLKEN0 |= MODCLKEN0_AUX_DDI0_OSC_EN; /*turn on oscillator interface clock */
+
+   /* configure HF clock */
+   #if HF_CLOCK_HF_ROSC
+     DDI_0_OSC->CTL0 |= 0x0; /* set 48MHz RCOSC */
+   #elif HF_CLOCK_HF_XOSC
+     DDI_0_OSC->CTL0 |= DDI_0_OSC_CTL0_SCLK_HF_SRC_SEL; /* set 48MHz XOSC (24MHz doubled internally) */
+   #endif
```

Comment:
the preprocessor branching is a bit ugly :-)
`DDI_0_OSC->CTL0 |= 0x0;` should be enough for the start.
we can think about making this configurable later.